fix: bugs with MetaDataset & random crop + add dependencies

zaptrem opened the pull request:
This PR fixes a few issues I ran into while setting things up with ZQEvans' dataset.

The dataloader could cause a crash in certain cases...
- When using `optimized_random_crop` with certain sample rates
- If a sample's audio was valid but ID3 tags were invalid.
- When mappings had not yet been generated
- When attempting to load a corrupt audio sample.
- When attempting to load an audio sample with `optimized_random_crop` disabled(?)

Additionally, while `pandas` and `yt_dlp` are required to import the library, they previously weren't listed in dependencies.

zaptrem committed:
Fix MetaDataset bugs and add dependencies

zaptrem committed:
Handle invalid ID3 & crops of some sample rates

zaptrem committed:
Fix corrupt audio loading crash and mypy types

flavioschneider commented:
Can you put the ID3 extension in a `WavID3Dataset` or something named like that, and the argument for the optimized random crop you can simply call it `random_crop`

zaptrem committed:
fix: rename optimized_random_crop & expose counts

zaptrem committed:
fix: update docs

zaptrem commented:
> Can you put the ID3 extension in a `WavID3Dataset` or something named like that, and the argument for the optimized random crop you can simply call it `random_crop`

I renamed `optimized_random_crop_size` to `random_crop_size`, but the ID3 extension will be difficult to move. I moved it there in the first place to handle the case where `torchaudio` is able to read a file but `TinyTag` isn't. Moving it back outside would require a complicated dance of calling back and fourth to and from the superclass's load functions and lots of catch logic that would complicate `MetaDataset` (but is a one-liner when in `WAVDataset`).
